### ISSUE TITLE
Update AST to distinguish `complex*8` from `complex(8)`

### ISSUE THREAD

**Issue by certik** [lfortran/lfortran#728]
Right now the two are represented by exactly the same AST, but they are not the same. `complex*8` is equivalent to `complex(4)`.

A test example:
```fortran
program complex3
complex*8 :: k
complex(8) :: z
print *, kind(k)
print *, kind(z)
end program
```
Which should return:
```console
gfortran a.f90 && ./a.out
           4
           8
```
But LFortran's AST represents both the same:
```console
$ lfortran --show-ast a.f90 --indent
(TranslationUnit 
    [(Program 
        complex3 
        () 
        [] 
        [] 
        [(Declaration 
            (AttrType 
                TypeComplex 
                [(() 
                8 
                Value)] 
                () 
                None
            ) 
            [] 
            [(k 
            [] 
            [] 
            () 
            None 
            ())] 
            ()
        )
        (Declaration 
            (AttrType 
                TypeComplex 
                [(() 
                8 
                Value)] 
                () 
                None
            ) 
            [] 
            [(z 
            [] 
            [] 
            () 
            None 
            ())] 
            ()
        )] 
        [(Print 
            0 
            () 
            [(FuncCallOrArray 
                kind 
                [] 
                [(() 
                k 
                ())] 
                [] 
                []
            )] 
            ()
        )
        (Print 
            0 
            () 
            [(FuncCallOrArray 
                kind 
                [] 
                [(() 
                z 
                ())] 
                [] 
                []
            )] 
            ()
        )] 
        []
    )]
)
```
So we have to fix this in the parser.

**Comment by lucifer1004** [comment]
Is this fixed already? I tested locally and the AST output was correct.

**Comment by certik** [comment]
I forgot to close this, this was fixed by #783.
